### ISSUE TITLE
Phone auth crashes on iOS

### ISSUE THREAD

**Issue by mudam** [dpa99c/cordova-plugin-firebasex#305]
# Bug report

**Current behavior:**

Crash on iOS when `signInWithCredential` is invoked. 

**Expected behavior:**
Should sign in.

**Steps to reproduce:**

Building app with Ionic (see details below). FirebasepluginX is second plugin to be added after the ones that come with ionic (first is geolocation). See the code (redacted) below what happens.

1. Invoke `verifyPhoneNumber` to obtain credential:
```
this.fbx.verifyPhoneNumber((credential: any) => {
    console.log(credential);
    this.verificationId = credential.verificationId;
  }, (err) => {
    reject(err);
    console.log(err);
  }, phoneNo, );
});
```
2. Log (first line in the method above) appears to be ok:
`2020-02-16 15:06:50.370350+0200 MyApp[1373:214189] {"verified":"false","verificationId":"AM5PThC5Y21NmJ9OTnGkiqQ-ywc_7p8VmcTSQMsilvaIAfBpvGV0MLfyVzCwp5ekqBhT4g_V9wlos9YE4XcQH1L0qSokfTA0m8T2ojWP0_X7lf71wfLlZW35mFEzPY3yIVC-tDIWt79IjSqrDvH-2cMfu_dTqrAXA5xSijA832Xk2KIJbxa1g8pBxEwdADLiVHAsSn-XfsEygNGM9ZLuY3qLpzaP6Puleg","instantVerification":"false"}`

3. Next we call :
```
signInWithCredential(code): Promise<void> {
    console.log('verid+pin', this.verificationId, code);
    return new Promise((resolve, reject) => {
      this.fbx.signInWithCredential(this.verificationId, code, function () {
        console.log("Successfully signed in");
        resolve();
      }, function (err) {
        console.error("Failed to sign in", err);
        reject(err);
      });
```
4. At this point the app crashes (SMS has been received if using real phone number). This happens when using either dummy number (set up in firebase) or real. XCode log:
```
2020-02-16 15:07:34.692055+0200 MyApp[1373:214189] verid+pin AM5PThC5Y21NmJ9OTnGkiqQ-ywc_7p8VmcTSQMsilvaIAfBpvGV0MLfyVzCwp5ekqBhT4g_V9wlos9YE4XcQH1L0qSokfTA0m8T2ojWP0_X7lf71wfLlZW35mFEzPY3yIVC-tDIWt79IjSqrDvH-2cMfu_dTqrAXA5xSijA832Xk2KIJbxa1g8pBxEwdADLiVHAsSn-XfsEygNGM9ZLuY3qLpzaP6Puleg 555555
2020-02-16 15:07:34.710409+0200 MyApp[1373:214189] [Process] kill() returned unexpected error 1
2020-02-16 15:07:34.711977+0200 MyApp[1373:214982] -[__NSCFNumber length]: unrecognized selector sent to instance 0xcfa0bfe97a082bb1
2020-02-16 15:07:34.712981+0200 MyApp[1373:214982] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFNumber length]: unrecognized selector sent to instance 0xcfa0bfe97a082bb1'
*** First throw call stack:
(0x190c3aa48 0x190961fa4 0x190b3e5a8 0x190c3eaf4 0x190c40a7c 0x104a72240 0x104a69144 0x104a68978 0x10561d7fc 0x10561ebd8 0x105625b48 0x1056266e4 0x105631adc 0x190956b88 0x190959760)
2020-02-16 15:07:34.714287+0200 MyApp[1373:214189] [Process] kill() returned unexpected error 1
libc++abi.dylib: terminating with uncaught exception of type NSException
```
**Environment information**
```
Ionic:

   Ionic CLI                     : 5.2.5 (/usr/local/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.11.5
   @angular-devkit/build-angular : 0.803.23
   @angular-devkit/schematics    : 8.1.3
   @angular/cli                  : 8.1.3
   @ionic/angular-toolkit        : 2.1.1

Cordova:

   Cordova CLI       : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms : android 8.1.0, ios 5.1.1
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 4.1.3, (and 8 other plugins)

Utility:

   cordova-res : 0.9.0 
   native-run  : 0.2.8 (update available: 0.3.0)

System:

   Android SDK Tools : 26.1.1 (/Users/madis/Library/Android/sdk/)
   ios-deploy        : 1.9.4
   ios-sim           : 8.0.2
   NodeJS            : v10.16.3 (/usr/local/Cellar/node@10/10.16.3/bin/node)
   npm               : 6.9.0
   OS                : macOS High Sierra
   Xcode             : Xcode 10.1 Build version 10B61
```

Package.json:
```
{
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/common": "~8.1.2",
    "@angular/core": "~8.1.2",
    "@angular/fire": "^5.2.3",
    "@angular/forms": "~8.1.2",
    "@angular/platform-browser": "~8.1.2",
    "@angular/platform-browser-dynamic": "~8.1.2",
    "@angular/router": "~8.1.2",
    "@ionic-native/core": "^5.0.0",
    "@ionic-native/firebase-x": "^5.21.5",
    "@ionic-native/geolocation": "^5.20.0",
    "@ionic-native/splash-screen": "^5.0.0",
    "@ionic-native/status-bar": "^5.0.0",
    "@ionic/angular": "^4.7.1",
    "@ngx-translate/core": "^11.0.1",
    "@ngx-translate/http-loader": "^4.0.0",
    "calendar-months": "^1.0.1",
    "cordova-android": "8.1.0",
    "cordova-ios": "5.1.1",
    "cordova-plugin-androidx": "^1.0.2",
    "cordova-plugin-androidx-adapter": "^1.1.0",
    "cordova-plugin-firebasex": "^7.0.2",
    "cordova-plugin-geolocation": "^4.0.2",
    "core-js": "^2.5.4",
    "firebase": "^7.4.0",
    "libphonenumber-js": "^1.7.39",
    "moment": "^2.24.0",
    "ngx-avatar": "^3.7.0",
    "rxjs": "~6.5.1",
    "tslib": "^1.9.0",
    "zone.js": "~0.9.1"
  },
  "devDependencies": {
    "@angular-devkit/architect": "~0.801.2",
    "@angular-devkit/build-angular": "^0.803.23",
    "@angular-devkit/core": "~8.1.2",
    "@angular-devkit/schematics": "~8.1.2",
    "@angular/cli": "~8.1.2",
    "@angular/compiler": "~8.1.2",
    "@angular/compiler-cli": "~8.1.2",
    "@angular/language-service": "~8.1.2",
    "@ionic/angular-toolkit": "^2.1.1",
    "@types/jasmine": "~3.3.8",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "~8.9.4",
    "codelyzer": "^5.0.0",
    "cordova-plugin-device": "^2.0.2",
    "cordova-plugin-ionic-keyboard": "^2.2.0",
    "cordova-plugin-ionic-webview": "^4.1.3",
    "cordova-plugin-splashscreen": "^5.0.2",
    "cordova-plugin-statusbar": "^2.4.2",
    "cordova-plugin-whitelist": "^1.3.3",
    "jasmine-core": "~3.4.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~4.1.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "~2.0.1",
    "karma-jasmine": "~2.0.1",
    "karma-jasmine-html-reporter": "^1.4.0",
    "protractor": "~5.4.0",
    "ts-node": "~7.0.0",
    "tslint": "~5.15.0",
    "typescript": "~3.4.3"
  },
  "description": "An Ionic project",
  "cordova": {
    "plugins": {
      "cordova-plugin-geolocation": {},
      "cordova-plugin-whitelist": {},
      "cordova-plugin-statusbar": {},
      "cordova-plugin-device": {},
      "cordova-plugin-splashscreen": {},
      "cordova-plugin-ionic-webview": {},
      "cordova-plugin-ionic-keyboard": {},
      "cordova-plugin-firebasex": {
        "FIREBASE_ANALYTICS_COLLECTION_ENABLED": "true",
        "FIREBASE_PERFORMANCE_COLLECTION_ENABLED": "true",
        "FIREBASE_CRASHLYTICS_COLLECTION_ENABLED": "true",
        "ANDROID_ICON_ACCENT": "#FF00FFFF",
        "ANDROID_PLAY_SERVICES_TAGMANAGER_VERSION": "17.0.0",
        "ANDROID_FIREBASE_ANALYTICS_VERSION": "17.2.1",
        "ANDROID_FIREBASE_MESSAGING_VERSION": "20.0.0",
        "ANDROID_FIREBASE_CONFIG_VERSION": "19.0.3",
        "ANDROID_FIREBASE_PERF_VERSION": "19.0.1",
        "ANDROID_FIREBASE_AUTH_VERSION": "19.1.0",
        "ANDROID_CRASHLYTICS_VERSION": "2.10.1",
        "ANDROID_CRASHLYTICS_NDK_VERSION": "2.1.1"
      }
    },
    "platforms": [
      "ios",
      "android"
    ]
  }
}
```

**Comment by dpa99c** [comment]
`verifyPhoneNumber()` has been reworked on the [dev branch](https://github.com/dpa99c/cordova-plugin-firebasex/tree/dev) as part of the effort to add support for Firebase Authentication (#208).

Please install the dev branch version directly until such time as this is merged to master and released via npm:

    cordova plugin add https://github.com/dpa99c/cordova-plugin-firebasex#dev

Please retest using the [example project - dev branch](https://github.com/dpa99c/cordova-plugin-firebasex/tree/dev) to rule out any causes in your own project.

**Comment by dpa99c** [comment]
The reworked authentication functionality has now been released in `cordova-plugin-firebasex@8.0.0` which should resolve this issue.
Please retest using the master branch of the [example project](https://github.com/dpa99c/cordova-plugin-firebasex).
